### DISCUSSION TITLE
Optimising rate limiter

### DISCUSSION THREAD

**Discussion by fdmackay** [joalla/discogs_client#102] [Support] [answered]
Thanks for this tool - I'm using it to extract marketplace stats for use in a data science course I'm taking. However I am finding the rate limiter seems to be adding excessive delays, such that 500 API calls take 950s. I've hardcoded a delay into my loop so each iteration takes as close to 1s as possible to avoid hitting the limit, but it would be nice to make this adjustable to account for the actual requests used/remaining - are the headers Discogs returns with this information available via the client?

**Comment by AnssiAhola** [comment, marked as answer]
Hey @fdmackay

You can disable the rate limiter with 
```python
client.backoff_enabled(false)
```
and you can access the rate limit headers with
```python
client._fethcer.rate_limit
client._fethcer.rate_limit_used
client._fethcer.rate_limit_remaining
```
_if I recall correctly_

**Reply by fdmackay** [reply]
Thanks!
